src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,181 @@
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE cart_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                ShoppingCart shoppingCart = getShoppingCartFromResultSet(resultSet);
+                statement.close();
+                shoppingCart.setProducts(
+                        getProductsFromShoppingCart(shoppingCart.getId(), connection));
+                return Optional.of(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get shopping cart from DB with id "
+                    + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT * FROM shopping_carts";
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart shoppingCart = getShoppingCartFromResultSet(resultSet);
+                shoppingCarts.add(shoppingCart);
+            }
+            statement.close();
+            for (ShoppingCart shoppingCart : shoppingCarts) {
+                shoppingCart.setProducts(
+                        List.copyOf(getProductsFromShoppingCart(shoppingCart.getId(), connection)));
+            }
+            return shoppingCarts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get shopping carts", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        String queryToDeleteProducts = "DELETE FROM shopping_carts_products WHERE cart_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToDeleteProducts)) {
+            statement.setLong(1, shoppingCart.getId());
+            statement.executeUpdate();
+            statement.close();
+            deleteShoppingCartFromCartsProducts(shoppingCart.getId(), connection);
+            insertProducts(shoppingCart, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete products from shopping cart "
+                    + shoppingCart, e);
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM shopping_carts WHERE cart_id = ?";
```

Comment:
1) Does it work? you are first deleting the shopping cart but only then referenced values(from shopping_carts_products). Probably you are using cascade(which automatically  delete all references before deleting the main object)  in workbench but then deleteShoppingCartFromCartsProducts() do nothing.
2) Let's update deleted column instead of deleting the entity